src/unix/fs.c

```diff
@@ -1251,22 +1271,39 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
       return UV__ERR(errno);
   }
 
-  flags = COPYFILE_ALL;
+  if ((req->flags & UV_FS_COPYFILE_FICLONE) ||
+      (req->flags & UV_FS_COPYFILE_FICLONE_FORCE)) {
+    if (!uv__fs_clonefile_mac(req))
+      return 0;  /* success */
 
-  if (req->flags & UV_FS_COPYFILE_FICLONE)
-    flags |= COPYFILE_CLONE;
 
-  if (req->flags & UV_FS_COPYFILE_FICLONE_FORCE)
-    flags |= COPYFILE_CLONE_FORCE;
+    /* If an error occurred and force was set, return the error to the caller;
+     * fall back to copyfile() when force was not set. */
+    if (req->flags & UV_FS_COPYFILE_FICLONE_FORCE)
+      return UV__ERR(errno);
```

Comment:
With my previous suggestion in mind:
```suggestion
      return rc;
```
Assign the result of `uv__fs_clonefile_mac()` to `rc` a few lines up and declare `rc` at the top of the function.